packages/shopping-cart/src/use-shopping-cart-manager.ts

```diff
@@ -188,6 +188,8 @@ export default function useShoppingCartManager( {
 		}
 	}, [ hookState.queuedActions, cacheStatus ] );
 
+	const responseCart = lastValidResponseCart.current;
```

Comment:
I'd personally move this to line 168, right after `lastValidResponseCart.current` is assigned. So that that entire computation is at one place. You could extract that computation to a custom hook:
```js
const responseCart = useLastValidCart( tempResponseCart, cacheStatus );
```
where the `useRef` and `useMemo` would be called inside that hook.

I'm not suggesting to actually do that, but demonstrating that it's possible and that the code belongs together.